lib/modules/manager/sbt/extract.ts

```diff
@@ -36,12 +41,16 @@ interface Ctx {
   variableName?: string;
 }
 
+const SBT_MVN_REPO = 'https://repo1.maven.org/maven2';
+
 const scala = lang.createLang('scala');
 
 const sbtVersionRegex = regEx(
   'sbt\\.version *= *(?<version>\\d+\\.\\d+\\.\\d+)',
 );
 
+const sbtProxyUrlRegex = regEx(/https?:.[a-z0-9/][^,]+/);
```

Comment:
```suggestion
const sbtProxyUrlRegex = regEx(/^\s*(?<repoName>\S+):\s+(?<proxy>https?:\/\/[\w./-]+)/);
```
Lets look also if there is a key, just to have a safe guard if there is a file named repositories with URLs.

Can the `repositories` file be located anywhere in the repository or has it to be located next to some of the other files?

Reply:
The repositories file can be under a .sbtopts folder or in a custom location configured with the property -`Dsbt.repository.config=<path-to-your-repo-file>`
Docs links:
https://www.scala-sbt.org/1.x/docs/Proxy-Repositories.html#sbt+Configuration
https://www.scala-sbt.org/1.x/docs/Proxy-Repositories.html#Launcher+Script